### PR TITLE
patch CodeInlineRenderer

### PR DESCRIPTION
**Pull request by fkpwolf**
[AB#337359](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/337359)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6795)

### CODE REVIEW

File: src/docfx/lib/markdown/MarkdownEngine.cs [hunk 1]
```diff
@@ -144,6 +145,30 @@ public string ToHtml(MarkdownObject markdownObject)
             return sb.ToString();
         }
 
+        // patch of CodeInlineRenderer, will be fixed in upstream. todo
```
Comment (yufeih) [inline src/docfx/lib/markdown/MarkdownEngine.cs:148]: consider put this class after all methods

File: docs/specs/validation/heading-validation.yml [hunk 2]
```diff
@@ -1323,6 +1323,44 @@ outputs:
     {"message_severity":"suggestion","log_item_type":"user","code":"duplicate-h1","message":"H1 'Title 1' is duplicated with other articles: 'a.md, b.md'","file":"b.md","line":1,"end_line":1,"column":1,"end_column":1}
     {"message_severity":"suggestion","log_item_type":"user","code":"duplicate-h1","message":"H1 'Title 1' is duplicated with other articles: 'a.md, b.md'","file":"a.md","line":1,"end_line":1,"column":1,"end_column":1}
 ---
+# Content validation - duplicated h2
```
Comment (yufeih) [inline docs/specs/validation/heading-validation.yml:1326]: nit: a simpler case would be to assert the title of

```
# `<`
```

is `<` in _markdown.yml_

**Review by yufeih — APPROVED** [review]
This is a quick fix on our side, can you help open a bug in https://github.com/lunet-io/markdig and and eventually fix it there?

File: docs/specs/validation/heading-validation.yml (hunk 2 above)
Comment (fkpwolf) [reply docs/specs/validation/heading-validation.yml:1326]: added

File: src/docfx/lib/markdown/MarkdownEngine.cs (hunk 1 above)
Comment (fkpwolf) [reply src/docfx/lib/markdown/MarkdownEngine.cs]: ok

Comment (fkpwolf) [reply src/docfx/lib/markdown/MarkdownEngine.cs]: https://github.com/lunet-io/markdig/issues/495

**Review by yufeih — APPROVED** [review]

**Pull request merged**